src/plugins/vis_augmenter/common/types.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ExpressionFunctionDefinition } from '../../expressions';
+
+export interface VisLayer {
+  // will be used as the column ID
+  id: string;
+  // will be used as the name when hovering over the tooltip
+  name: string;
+}
+
+export type VisLayers = VisLayer[];
+
+// resourceId & resourceName are required so that the
+// events flyout can partition data based on these attributes
+// (e.g., partitioning anomalies based on the detector they came from)
+export interface PointInTimeEventMetadata {
+  resourceId: string;
+  resourceName: string;
+  tooltip?: string;
+}
+
+export interface PointInTimeEvent {
+  timestamp: number;
+  metadata: PointInTimeEventMetadata;
+}
+
+export interface PointInTimeEventsVisLayer extends VisLayer {
+  events: PointInTimeEvent[];
+}
+
+// used to determine what vis layer's interface is being implemented.
+// currently PointInTimeEventsLayer is the only interface extending VisLayer
+export const isPointInTimeEventsVisLayer = (obj: any) => {
+  return 'events' in obj;
+};
```

Comment:
How are you imagining this to be used? It seems like it would be simpler to have a static `type` string on the `VisLayer` interface that you can specify when defining a particular type of `VisLayer`, and then just access as a value. This type of convenient method may still be helpful, but it's actually implemented as more a capability check.

Reply:
Agreed, existing soln seemed hacky. Updated to use enums in latest commit. Also added a general `isValidVisLayer` fn that can be ran when verifying the new saved object types are valid or not, based on expression function type